packages/components/checkbox/src/Checkbox.tsx

```diff
@@ -22,41 +24,60 @@ export const Checkbox = forwardRef<HTMLButtonElement, CheckboxProps>(
     },
     forwardedRef
   ) => {
+    const innerId = useId()
+    const innerLabelId = useId()
+
     const field = useFormFieldState()
     const group = useCheckboxGroup()
+
     const rootRef = useRef<HTMLButtonElement | undefined>()
     const ref = useMergeRefs(forwardedRef, rootRef)
 
-    const name = field.name ?? group.name
-    const isRequired = field.isRequired ?? group.isRequired
-    const isInvalid = field.isInvalid ?? group.isInvalid
-    const isFieldEnclosed = field.id !== group.id
-    const id = isFieldEnclosed ? field.id : undefined
-    const description = isFieldEnclosed ? field.description : undefined
+    const getCheckboxAttributes = ({
```

Comment:
defining this function outside of the `Checkbox` component would simplify it even further imho